npm/private/npm_translate_lock_state.bzl

```diff
@@ -56,11 +59,11 @@ WARNING: `update_pnpm_lock` attribute in `npm_translate_lock(name = "{rctx_name}
     if pnpm_lock_exists:
         _load_lockfile(priv, rctx, label_store)
 
-    if not root_package_json_declared:
-        # If a root package.json label wasn't declared then we didn't load pnpm.patchedDependencies
-        # above. Try again now that the lockfile has been read so that we can derive the root
-        # package.json from the `importers` field.
-        _init_patched_dependencies_labels(priv, rctx, label_store)
+    # If we didn't initialize the patched dependency labels above then try again now. We'll either
```

Comment:
"if we didn't ... then try again" - don't we try again every time now?

Reply:
I simplified the logic so the function itself marks itself as called already and exits early

Reply:
Should the comment change then?

Reply:
Sure. I'll re-phrase so it is more obvious

Reply:
Actually, I'll move the "already intialized" state to this function so it reads clearly

Reply:
> Actually, I'll move the "already intialized" state to this function so it reads clearly

sgtm 👍